doc/dependency_management.rst

```diff
@@ -0,0 +1,138 @@
+``LIB_DEPENDENT_MODULES`` variables define the sub-dependency relationships. This builds up a tree which is
+traversed depth-first to populate the sandbox.
+
+As an example, suppose that an application's ``CMakLists.txt`` contains ``set(APP_DEPENDENT_MODULES lib_mod0 lib_mod1)``
+and then the modules have the following in their ``lib_build_info.cmake`` files:
+
+========  ====================================================
+lib_mod0  ``set(LIB_DEPENDENT_MODULES "lib_mod2")``
+lib_mod1  ``set(LIB_DEPENDENT_MODULES "lib_mod2" "lib_mod3")``
+lib_mod2  ``set(LIB_DEPENDENT_MODULES "lib_mod3")``
+lib_mod3  ``set(LIB_DEPENDENT_MODULES "")``
+========  ====================================================
+
+Then the dependent modules will be retrieved in the following order: ``lib_mod0``, ``lib_mod2``, ``lib_mod3``, ``lib_mod1``.
+
+If a dependency is not present in the sandbox, it will be retrieved the first time it is traversed in this tree.
+If it then appears again as a dependency of another module, nothing will happen because it is already present
+in the sandbox.
+
+The ``APP_DEPENDENT_MODULES`` and ``LIB_DEPENDENT_MODULES`` variables define lists of dependencies, where each
+element in the list is a string that specifies where to fetch the source code from and which version to fetch.
+See the :ref:`dependency-format` specification for full details of the accepted format.
+
+Retrieval happens by cloning from a git repo, which can be accessed via HTTPS (for public repositories) or via
+an SSH key if one is configured for passwordless access.
+
+All the dependencies in the tree will be retrieved into the sandbox, if they are not already present. The location
+of the root of the sandbox must be specified by the application using the ``XMOS_SANDBOX_DIR`` variable. If an
+application or static library has no dependencies, this variable doesn't need to be set.
+
+During the process of dependency resolution, if a module is already present in the sandbox it will not be modified.
+If a different version of a module is subsequently required, the procedure is to use standard ``git`` commands to
+change to the desired version and then run ``cmake build`` in the application directory.
+
+Sandbox Manifest
+^^^^^^^^^^^^^^^^
+
+Whenever ``cmake`` generates the build environment for an application or static library, a file called ``manifest.txt``
+is created in the build directory. This file contains a list of the application and module dependencies, with the columns in
+a table for the location from which the module was cloned, and the branch/tag and git commit hash for the current checked
+out changeset. If any columns are not applicable, they will contain a hyphen.
+
+Dependency Location
+^^^^^^^^^^^^^^^^^^^
+
+By default, the location of dependent modules is the root of the sandbox as defined by the ``XMOS_SANDBOX_DIR``
+variable. This can be overridden on a per-dependency basis by setting a variable for each non-default dependency
+location.
+
```

Comment:
Add a ..note:: that this is the recommend behaviour

Reply:
Done - this note directive is useful, I hadn't seen it before.